changelog/20.0/20.0.0/summary.md

```diff
@@ -93,6 +94,71 @@ Vitess' endtoend tests no longer use nor test `gh-ost` migrations.
 
 ### <a id="breaking-changes"/>Breaking Changes
 
+#### <a id="enum-set-vstream"/>ENUM and SET column handling in VTGate VStream API
+
+The [VTGate VStream API](https://vitess.io/docs/reference/vreplication/vstream/) now returns [`ENUM`](https://dev.mysql.com/doc/refman/en/enum.html) and [`SET`](https://dev.mysql.com/doc/refman/en/set.html) column type values in [`VEvent`](https://pkg.go.dev/vitess.io/vitess/go/vt/proto/binlogdata#VEvent) messages (in the embedded [`RowChange`](https://pkg.go.dev/vitess.io/vitess/go/vt/proto/binlogdata#RowChange) messages) as their string values instead of the integer based ones — in both the copy/snapshot phase and the streaming phase. This change was done to make the `VStream` API more user-friendly, intuitive, and to align the behavior across both phases. Before [this change](https://github.com/vitessio/vitess/pull/15723) the values for [`ENUM`](https://dev.mysql.com/doc/refman/en/enum.html) and [`SET`](https://dev.mysql.com/doc/refman/en/set.html) columns were string values in the copy phase but integer values (which only have an internal meaning to MySQL) in the streaming phase. This inconsistency led to various [challenges and issues](https://github.com/vitessio/vitess/issues/15750) for each `VStream` client/consumer (e.g. the [`Debezium` Vitess connector](https://debezium.io/documentation/reference/stable/connectors/vitess.html) failed to properly perform a snapshot for tables containing these column types). Now the behavior is intuitive — clients need the string values as the eventual sink is often not MySQL so each consumer needed to perform the mappings themselves — and consistent. While this is a (potentially) breaking change, a new boolean field has been added to the [`FieldEvent`](https://pkg.go.dev/vitess.io/vitess/go/vt/proto/binlogdata#FieldEvent) message called `EnumSetStringValues`. When that value is `false` then the consumer will need to perform the mappings as before (in Vitess v19 and older), but when this field is `true` then no mapping is required. This will help to ensure a smooth transition for all consumers over time. To demonstrate, let's look at the textual output (printing the received `VEvents` as strings) when streaming a single `enum_set_test` table from an unsharded keyspace so that we can see what the VStream looks like before and after when we start a new VSTream in copy/snapshot mode and then transition to streaming mode for the following table:
```

Comment:
I think we should make it clear in the following sentence that the mappings are only necessary in streaming not in copy phase.

```
When that value is `false` then the consumer will need to perform the mappings as before (in Vitess v19 and older), but when this field is `true` then no mapping is required.
```

for example change to:

```
When that value is `false` (in Vitess v19 and older) then the consumer will need to perform the mappings during streaming phase, but not during copy phase. When this field is `true`, then no mapping is required.
```

Reply:
Thanks, @twthorn ! I've addressed that in https://github.com/vitessio/vitess/pull/15723/commits/0c98605bbdd696579394ee89b2584bdb5fc82cbf

Reply:
Thanks, @twthorn ! I've addressed that in https://github.com/vitessio/vitess/pull/15723/commits/edf9fae7ca86b17bb2832e241a2514686680226a